address/address.go

```diff
@@ -78,6 +78,12 @@ const (
 	TapScriptVersion uint8 = 0
 )
 
+// ProofCourierAddr is the address of the proof courier that will be used to
+// distribute related proofs for this address.
+//
+// NOTE: See the proof package for more details.
+type ProofCourierAddr []byte
```

Comment:
If we're going with the URI approach, then should this actually be a `string` for the purposes of internal representation? 

We may also want to consider the tradeoffs of the custom URI approach vs either the type for each courier type appraoch or the type prefix approach. 

Nice thing is for the URI approach, we _should_ be able to re-use this method from the stdlib: https://pkg.go.dev/net/url#ParseRequestURI. One other thing to consider is how we track the set of URIs we know about, and/or mark those as required vs not.

Reply:
I've changed this to a `url.URL` type.